main/main.c

```diff
@@ -102,6 +111,35 @@ parse_opt(int key, char *arg, struct argp_state *state)
 		args->log_base_dir = arg;
 		break;
 
+	case 'm': {
+		long log_file_mode;
+		char *end;
+
+		/* Assuming the file mode is in octal. */
+		log_file_mode = strtol(arg, &end, 8);
+		if (arg == end || !*arg || *end) {
+			G_LOG(ERR,
+				"gatekeeper: log file mode \"%s\" is not a number\n",
+				arg);
+			return ARGP_ERR_UNKNOWN;
+		}
+		if ((log_file_mode == LONG_MAX || log_file_mode == LONG_MIN) &&
+				errno == ERANGE) {
+			G_LOG(ERR,
+				"gatekeeper: log file mode \"%s\" caused underflow or overflow\n",
+				arg);
```

Comment:
Better error message: `argp_error(state, "the log file mode \"%s\" is out of range", arg);`